class-es-wp-tax-query.php

```diff
@@ -309,50 +307,56 @@ public function transform_query( &$query, $resulting_field ) {
 
 		$resulting_field = sanitize_key( $resulting_field );
 
-		// phpcs:disable WordPress.DB.PreparedSQL.NotPrepared, WordPress.VIP.DirectDatabaseQuery.NoCaching, WordPress.VIP.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching, WordPress.DB.DirectDatabaseQuery.DirectQuery
+		// Empty 'terms' always results in a null transformation.
+		$terms = array_filter( $query['terms'] );
```

Comment:
Recommend wrapping this in `array_values` so it's properly zero-indexed:
```suggestion
		$terms = array_values( array_filter( $query['terms'] ) );
```